app/src/test/java/com/github/multimatum_team/multimatum/DeadlineDetailsTest.kt

```diff
@@ -9,28 +9,54 @@ import androidx.test.espresso.matcher.ViewMatchers.withText
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import com.github.multimatum_team.multimatum.model.Deadline
 import com.github.multimatum_team.multimatum.model.DeadlineState
+import com.github.multimatum_team.multimatum.service.ClockService
+import com.github.multimatum_team.multimatum.util.MockClockService
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import dagger.hilt.components.SingletonComponent
+import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 import java.time.LocalDate
+import java.time.LocalDateTime
+import javax.inject.Inject
 
 
 /**
  * Tests for the DeadlineDetailsActivity class
  */
 @RunWith(AndroidJUnit4::class)
+@HiltAndroidTest
+@UninstallModules(ClockModule::class)
 class DeadlineDetailsTest {
+    @get:Rule
+    var hiltRule = HiltAndroidRule(this)
+
+    @Inject
+    lateinit var clockService: ClockService
+
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        hiltRule.inject()
+    }
 
     @Test
     fun `Given a deadline not yet due or done, the activity should display it`() {
         val intent = DeadlineDetailsActivity.newIntent(
             ApplicationProvider.getApplicationContext(),
             "1",
-            Deadline("Test 1", DeadlineState.TODO, LocalDate.now().plusDays(7))
+            Deadline("Test 1", DeadlineState.TODO, clockService.now().plusDays(7))
```

Comment:
I am a bit confused about how this test (and more generally all the tests using `clockService.now()`) work. Do they really use the time at which the test is run, or is there some mocking to avoid that? Because in the former case the tests may pass or fail depending on when they are run, which is not acceptable.

Reply:
there's some mocking, it's the purpose of the existence of clockService. It allows us to override now() function to get a special fixed time for tests.